File: Solution/others/Java/SahilK/Testing/Cucumber/BDDCucumber/src/main/java/com/example/catalog/service/ProductService.java
package com.example.catalog.service;

import java.util.List;

import com.example.catalog.model.Product;

public interface ProductService {

  public List<Product> getAllList();
  public Product getProductById(int id);
  public Product createProduct(Product product);
  public Product updateProduct(Product product);
  public String deleteProduct(int id);
    
} 

File: Solution/Backend/python/webapi/products/services/product_service.py
from ..entities.product import Product


class ProductService:
    def __init__(self, repository):
        self.repository = repository

    def get_all_products(self):
        return self.repository.get_all()

    def get_product_by_id(self, id):
        return self.repository.get_by_id(id)

    def create_product(self, product_data):
        product = Product.from_dict(product_data)
        return self.repository.create(product)

    def update_product(self, id, product_data):
        product = Product.from_dict(product_data)
        return self.repository.update(id, product)

    def delete_product(self, id):
        self.repository.delete(id)
